File: src/types/lifecycle.ts
export interface ValidationResult {
  traceId: string;
  success: boolean;
  feedback?: string;
}

export interface LifecycleStats {
  captured: number;
  validated: number;
  synthesized: number;
  antiPatterns: number;
  pruned: number;
  total: number;
}

export interface SynthesizeOptions {
  minSuccessCount?: number;
}

export interface PruneOptions {
  minFailureCount?: number;
}

export const DEFAULT_MIN_SUCCESS_COUNT = 3;
export const DEFAULT_MIN_FAILURE_COUNT = 2;

File: python-sdk/src/graphmind_context_graphs/types/lifecycle.py
from __future__ import annotations
from dataclasses import dataclass

DEFAULT_MIN_SUCCESS_COUNT = 3
DEFAULT_MIN_FAILURE_COUNT = 2


@dataclass
class ValidationResult:
    trace_id: str
    success: bool
    feedback: str | None = None


@dataclass
class LifecycleStats:
    captured: int = 0
    validated: int = 0
    synthesized: int = 0
    anti_patterns: int = 0
    pruned: int = 0
    total: int = 0
